Sources/Confidence/Http/NetworkClient.swift

```diff
@@ -5,25 +5,29 @@ final class NetworkClient: HttpClient {
     private let retry: Retry
     private let session: URLSession
     private let baseUrl: String
+    private var timeoutIntervalForRequests: Double
 
     public init(
         session: URLSession? = nil,
         baseUrl: String,
         defaultHeaders: [String: String] = [:],
-        retry: Retry = .none
+        retry: Retry = .none,
+        timeoutIntervalForRequests: Double
     ) {
         self.session =
         session
         ?? {
             let configuration = URLSessionConfiguration.default
             configuration.httpAdditionalHeaders = defaultHeaders
+            configuration.timeoutIntervalForRequest = timeoutIntervalForRequests
 
             return URLSession(configuration: configuration)
         }()
 
         self.headers = defaultHeaders
         self.retry = retry
         self.baseUrl = baseUrl
+        self.timeoutIntervalForRequests = timeoutIntervalForRequests
```

Comment:
do we need to still store this in the class? or just setting it above in the session configuration is enough?

Reply:
I think setting in the session config should be enough, you're right.